packages/tools/fetch-tool/src/fluidFetchMessages.ts

```diff
@@ -140,16 +191,28 @@ async function* saveOps(
     // Split into 100K ops
     const chunk = 100 * 1000;
 
+    let sequencedMessages: ISequencedDocumentMessage[] = [];
+    // current sequence number to read from
+    let curr = 1;
     // Figure out first file we want to write to
     let index = 0;
     if (files.length !== 0) {
-        index = (files.length - 1);
+        index = files.length - 1;
+        const name = filenameFromIndex(index);
+        const fileContent = fs.readFileSync(`${dir}/messages${name}.json`, { encoding: "utf-8" });
+        const messages: ISequencedDocumentMessage[] = JSON.parse(fileContent);
+        // get current sequence number from the last op read, and plus one
+        curr = messages[messages.length - 1].sequenceNumber + 1;
```

Comment:
This is wrong. It should always be the first message not the last otherwise you will end up storing more than 100K ops in that file. Because letss say it already had 50k ops and then you started with last op and added 100k more ops, then it would have 150k ops.

Reply:
i was trying to avoid the situation that, if a file already had 100k ops, then it wouldn't need to write it again, it will just skip to the next file. i can add a condition

Reply:
This is a very rare condition with 1/100k chance and below code handles that. You can avoid handling this separately as it will make more complicated to read imo.

Reply:
Its upto you, you can add that condition.

Reply:
ok, just when I was testing, if `curr = messages[0].sequenceNumber + 1` and i set `chunk` to be 1000, i had 2500 ops and I deleted messages2.json, keeping messages.json and messages1.json. when i ran fetch tool again, it was writing both messages1.json and messages2.json

Reply:
so below code did not handle it

Reply:
Yes I meant that it will write 1 file again but chance is just 1/100K. So its a rare condition. What I meant was that it will not break the tool.